netlogo-gui/src/main/app/NotificationBanner.scala

```diff
@@ -0,0 +1,218 @@
+    }
+
+      val parser = new JSONParser()
+      val jsonArray = parser.parse(jsonContent).asInstanceOf[JSONArray]
+      jsonArray.toArray.flatMap { obj =>
+        val jsonObject = obj.asInstanceOf[JSONObject]
+        Option(jsonObject.get("event-id").asInstanceOf[Long].toInt).map { eventId =>
+          val title = Option(jsonObject.get("title")).map(_.toString).getOrElse("")
+          val fullText = Option(jsonObject.get("fullText")).map(_.toString).getOrElse("")
+          val date = Option(jsonObject.get("date")).map(_.toString).getOrElse("")
+          JsonObject(eventId, date, title, fullText)
+        }
+      }.toList.sortBy(_.eventId)(Ordering[Int].reverse)
+
+  }
+
+  // Method to show JSON content in a dialog
+  private def showJsonInDialog(): Unit = {
+
+    val prefs = Preferences.userNodeForPackage(getClass)
+    try {
+      val jsonContent = fetchJsonFromUrl()
+      jsonObjectList = parseJsonToList(jsonContent) // Populate the class variable
+      val formattedString = formatJsonObjectList(jsonObjectList)
+      if(!isShowNeeded()){
+        return
+      }
+      val lastSeenEventId = prefs.getInt(lastSeenEventIdKey, -1); // Returns -1 if "event-id" is not found
+      if(jsonObjectList.head.eventId > lastSeenEventId){
+        println(s"Show this ${jsonObjectList.head.eventId}")
+      }
+      else {
+        println(s"Don't show this  ${jsonObjectList.head.eventId}")
+        return
+      }
+      val html = InfoFormatter.toInnerHtml(formattedString)
+
+      if (!jsonContent.trim.isEmpty) {
+          editorPane = new JEditorPane {
+            self =>
+
+            setDragEnabled(false)
+            setEditable(false)
+            setContentType("text/html")
+            setOpaque(true)
+            setBackground(InterfaceColors.CODE_BACKGROUND) // Set the background color)
+            setForeground(InterfaceColors.DEFAULT_COLOR) //Set the font color
+            setText(html)
+            setCaretPosition(0)
+          }
+
+          val scrollPane = new JScrollPane(editorPane)
+          scrollPane.setPreferredSize(new Dimension(500, 400))
+          val panel = new JPanel(new BorderLayout())
+          panel.add(scrollPane, BorderLayout.CENTER)
+          val options: Array[AnyRef] = Array(I18N.gui.get("common.buttons.ok"))
+
+          val result = JOptionPane.showOptionDialog(
+            null, panel, I18N.gui.get("dialog.interface.newsNotificationTitle"),
+            JOptionPane.DEFAULT_OPTION, JOptionPane.PLAIN_MESSAGE, null, options, options(0)
+          )
+
+          // Check if OK was clicked (index 0 in options array) and hide the NotificationBanner
+          if (result == 0) {
+            setVisible(false) // Hide NotificationBanner
+            prefs.putInt("lastSeenEventId", jsonObjectList.head.eventId)
+          }
+      }
+    } catch {
+      case e: Exception =>{
+        org.nlogo.swing.OptionDialog.showCustom(this, I18N.gui.get("error.dialog.unknown"), e.getMessage, null)
```

Comment:
See above style guide comment, `OptionDialog` should be imported at the top and the full path should be removed.

Reply:
Done now as 
import org.nlogo.swing.{HoverDecoration, OptionDialog}